### PR TITLE
[DAT]-7422 Extend test-harness to be able to validate DML change types

### PR DESCRIPTION
**Pull request by AlexanderSashchenko**
Initial realization to review and decide

### CODE REVIEW

File: src/main/groovy/liquibase/harness/BaseHarnessSuite.groovy [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
 
 import liquibase.harness.change.ChangeObjectTests
+import liquibase.harness.data.ChangeDataTest
 import liquibase.harness.snapshot.SnapshotObjectTests;
 import org.junit.runner.RunWith;
 import org.junit.runners.Suite;
 
 @RunWith(Suite.class)
-@Suite.SuiteClasses([ChangeObjectTests, SnapshotObjectTests])
+@Suite.SuiteClasses([ChangeDataTest, ChangeObjectTests, SnapshotObjectTests])
```
Comment (KushnirykOleh) [inline src/main/groovy/liquibase/harness/BaseHarnessSuite.groovy:10]: nit
Other our test classes end with [s] what indicates it's not one test but actually number of them based on input

File: src/main/groovy/liquibase/harness/data/ChangeDataTest.groovy [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+import static liquibase.harness.util.FileUtils.*
+import static liquibase.harness.util.TestUtils.*
+import static ChangeDataTestHelper.*
+
+class ChangeDataTest extends Specification {
+
+    @Unroll
+    def "apply #testInput.changeData against #testInput.databaseName, #testInput.version; verify generated query, checking query and obtained result set"() {
+
+        Liquibase liquibase = createLiquibase(testInput.pathToChangeLogFile, testInput.database)
```
Comment (KushnirykOleh) [inline src/main/groovy/liquibase/harness/data/ChangeDataTest.groovy:22]: nit. Move [given:] one line up, before creating Liquibase

File: src/main/groovy/liquibase/harness/data/ChangeDataTest.groovy [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        assert checkingSql != null: "No checkingSql for ${testInput.changeData} against " +
+                "${testInput.database.shortName} " +
+                "${testInput.database.databaseMajorVersion}.${testInput.database.databaseMinorVersion}"
+
+        when: "get sql that is generated for changeset"
+        def generatedSql = cleanSql(toSqlFromLiquibaseChangeSets(liquibase))
+
+        then: "verify expected sql matches generated sql"
+        if (expectedSql != null && !testInput.pathToChangeLogFile.endsWith(".sql")) {
+            assert generatedSql == expectedSql: "Expected SQL does not match actual sql. " +
```
Comment (KushnirykOleh) [inline src/main/groovy/liquibase/harness/data/ChangeDataTest.groovy:49]: nit
mixing uppercase and lowercase "SQL"

File: src/main/groovy/liquibase/harness/data/ChangeDataTest.groovy [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+            println "Error executing sql. If this is expected to be invalid SQL for this database/version, " +
+                    "create an 'expectedSql/${testInput.database.shortName}/${testInput.changeData}.sql' file that starts with " +
+                    "'INVALID TEST' and an explanation of why."
+            throwable.printStackTrace()
+            Assert.fail throwable.message
+        }
+
+        then: "obtain resultSet form the statement, compare expected resultSet to generated resultSet, apply rollback"
+        try {
+            def statement = connection.createStatement()
```
Comment (KushnirykOleh) [inline src/main/groovy/liquibase/harness/data/ChangeDataTest.groovy:75]: nit
redundant variables. If they are used ones, inline subsequent calls. Check refactoring techniques for references

**Review by KushnirykOleh — CHANGES_REQUESTED** [review]
few nits, otherwise looks good

File: src/main/groovy/liquibase/harness/data/ChangeDataTest.groovy (hunk 2 above)
Comment (AlexanderSashchenko) [reply src/main/groovy/liquibase/harness/data/ChangeDataTest.groovy]: ++

File: src/main/groovy/liquibase/harness/data/ChangeDataTest.groovy (hunk 3 above)
Comment (AlexanderSashchenko) [reply src/main/groovy/liquibase/harness/data/ChangeDataTest.groovy]: ++

File: src/main/groovy/liquibase/harness/data/ChangeDataTest.groovy (hunk 4 above)
Comment (AlexanderSashchenko) [reply src/main/groovy/liquibase/harness/data/ChangeDataTest.groovy]: ++

**Review by KushnirykOleh — APPROVED** [review]

**Pull request merged**